community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/IndexSpecifier.java

```diff
@@ -58,21 +59,28 @@ public String toString()
         return specification;
     }
 
-    private Pair<String, String> parse()
+    private Pair<String,String[]> parse()
     {
-        Pattern pattern = Pattern.compile( "" +
-                ":" + or( simpleIdentifier( "simpleLabel" ), complexIdentifier( "complexLabel" ) ) +
-                "\\(" + or( simpleIdentifier( "simpleProperty" ), complexIdentifier( "complexProperty" ) ) + "\\)"
+        // Note that this now matches all properties in a single group, in order to split them later.
+        Pattern pattern = Pattern.compile(
+                ":" + or( identifier(true), qoutedIdentifier(true) ) + // Match the label
+                "\\((" + or( identifier(false), qoutedIdentifier(false) ) + // Match the first property
+                "(?:\\,\\s" + or( identifier(false), qoutedIdentifier(false) ) + ")*)\\)" // Match following properties
         );
         Matcher matcher = pattern.matcher( specification );
         if ( !matcher.find() )
         {
             throw new IllegalArgumentException( "Cannot parse index specification " + specification );
         }
-        return Pair.of(
-                either( matcher.group( "simpleLabel" ), matcher.group( "complexLabel" ) ),
-                either( matcher.group( "simpleProperty" ), matcher.group( "complexProperty" ) )
-        );
+        String label = either( matcher.group( 1 ), matcher.group( 2 ) );
+        String propertyString = matcher.group( 3 );
+        //Split string on commas, but ignore commas in quotes
+        String[] properties = propertyString.split(",\\s(?=(?:`[^`]*`)|[^`]*$)", -1);
```

Comment:
Likewise here you probably want a `*` after the `\\s`.
Also note that according to open Cypher, backticks can be escaped with double backticks.
I also don't see a test that involves a comma in a quoted identifier.
And why the `-1` instead of `0` to `String.split`?